File: java/src/test/java/sammancoaching/FakePercentages.java
package sammancoaching;

import java.util.HashMap;

public class FakePercentages extends SalaryContributionPercentages {
    private final HashMap<String, Double> salaryPercentages;

    public static FakePercentages getStandardValues() {
        var salaryPercentages = new HashMap<String, Double>();
        salaryPercentages.put("LONG_TENURE_PERCENTAGE", 3.5);
        salaryPercentages.put("MEDIUM_TENURE_PERCENTAGE", 2.0);
        salaryPercentages.put("NO_TENURE_PERCENTAGE", 0.0);
        salaryPercentages.put("LEADERSHIP_TEAM_PERCENTAGE", 2.5);
        salaryPercentages.put("MID_SENIORITY_PERCENTAGE", 3.0);
        salaryPercentages.put("BASE_CONTRIBUTION_RATE", 5.0);

        return new FakePercentages(salaryPercentages);
    }

    public FakePercentages(HashMap<String, Double> salaryPercentages) {
        super(null);
        this.salaryPercentages = salaryPercentages;
    }

    @Override
    public double lookupValue(String namedConstant) {
        return salaryPercentages.get(namedConstant);
    }
}

File: cpp/src/FakePercentages.cpp
#include "FakePercentages.h"


FakePercentages FakePercentages::getStandardValues() {
    std::unordered_map<std::string, double> salaryPercentages;
    salaryPercentages["LONG_TENURE_PERCENTAGE"] = 3.5;
    salaryPercentages["MEDIUM_TENURE_PERCENTAGE"] = 2.0;
    salaryPercentages["NO_TENURE_PERCENTAGE"] = 0.0;
    salaryPercentages["LEADERSHIP_TEAM_PERCENTAGE"] = 2.5;
    salaryPercentages["MID_SENIORITY_PERCENTAGE"] = 3.0;
    salaryPercentages["BASE_CONTRIBUTION_RATE"] = 5.0;

    return FakePercentages(salaryPercentages);
}

FakePercentages::FakePercentages(const std::unordered_map<std::string, double> &salaryPercentages)
    : SalaryContributionPercentages(nullptr), // Passing nullptr (like Java's null)
      salaryPercentages(salaryPercentages) {
}

double FakePercentages::lookupValue(const std::string &namedConstant) const {
    return salaryPercentages.at(namedConstant);
}
